docs/tech-reference/yum-plugins.rst

```diff
@@ -692,3 +714,45 @@ To test the generation of a .asc file::
     $ touch test
     $ gpg --yes --homedir /path/to/gnupg --detach-sign --armor test
     $ rm test test.asc
+
+**WARNING!** The example, as presented above, is not suitable for production
+use. Unprotected GPG keys may be easily stolen. You may want to consider
+more secure alternatives for your signing needs, like a dedicated server,
+potentially with a
+`Hardware Security Module <https://en.wikipedia.org/wiki/Hardware_security_module>`.
+
+GPG Signing Of Repository Metadata With An Alternate Command
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In certain circumstances, using ``/usr/bin/gpg`` is not acceptable
+as a signing solution, given that private keys cannot be passphrase-protected.
+
+If a different signing command is necessary, one can set the ``gpg_cmd``
+configuration variable to point to such command.
+
+The signing command will be passed the following environment variables:
+* ``GPG_CMD``
+* ``GPG_KEY_ID`` (if specified in the configuration file)
```

Comment:
This says "the configuration file", but the key ID can also be stored on the yum distributor object, correct?

Reply:
Yes, that is correct. All these can be done via distributor configurations as well, although GPG_CMD is probably not hugely helpful. GPG_KEY_ID is.

Would you like me to update the documentation to include this?

Also, is it possible for this to be included in pulp 2.16? I saw an email mentioning that it would need to be merged today.

Reply:
I'm commenting as an interested user only.  We would use this feature, but we'd probably set the key ID on each repo's distributor object.

If written as, "if specified in the configuration", i.e. drop the "file", then I'd have understood it as meaning all of the usual ways config can be provided to a distributor: on the distributor itself, on the config file for that distributor type, or in override_config during a call to publish.

Reply:
Agreed, and fixed.